Mod2/Lessons/Week2/DataRelationships.md

```diff
@@ -0,0 +1,158 @@
+
+But what about strings? They can vary in length. The Database handles strings by giving them a maximum length, which is 255 bytes by default. If we tried to do the same thing with arrays, we would be limiting how many relationships we could create, which is a bigger problem than limiting how long a string can be.
+
+The key takeaway here is a database can’t store an array of ids, so we’re going to need something else.
+
+
+## Join Tables
+Since we can’t achieve the many-to-many relationship with our given tables, we are going to add a third table to manage this relationship. This is called a join table.
+
+<p align='center'>
+  <img src='../../Images/Week2/many_to_many_good.png'>
+</p>
+
+Note: Join tables are just ordinary tables with a unique purpose. Each row of our join table relates a row of one table to a row of another table.
+
+Turn and Talk: Which songs are associated with which playlists?
+
+The join table’s name doesn’t really matter. We could call it Song_Playlist or Playlist_Song, it’s really up to you as the developer. You could even choose to name it happy_fun_times but that would be confusing.
+
+When you’re thinking about what to call this table, think about how you’re likely to use it most within your application. Since our app’s goal will be to show a playlist of songs more often, we’re going to call it playlist_songs.
+
+Note: Don’t confuse join table with a sql joins operation. They are two different things.
+
+
+## ERD Practice
+
+These diagrams we've been drawing to visualize data relationships are called Entity Relationship Diagrams, ERDs for short. Let's take a look at a larger ERD.
+
+<p align='center'>
+  <img src='../../Images/Week2/car_accident_ERD.png'>
+</p>
+
+Annotate this diagram as much as you can! What type of relationships do you see? Do you see any primary or foreign keys?
+
+Then, write sentences to describe these relationships.
+
+<details><summary>Annotated Car Accident ERD</summary><br/>
+
+<p align='center'>
+  <img src='../../Images/Week2/car_accident_ERD_annotated.png'>
+</p>
+</details>
+</br>
+
+This ERD is from the [Visual Paradigm](https://circle.visual-paradigm.com/category/system-design-development/entity-relationship-diagram/) website. They have many more examples if you want more practice!
+
+There are more nuanced symbols you can use, such as the open circle. [Here](https://vertabelo.com/blog/crow-s-foot-notation/) is a blog post if you want to learn more.
+
+
+## Checks for Understanding
+
+1. How is a one-to-many relationship set up in a database?
+1. How would you describe the one to many relationship between people and toothbrushes?
+1. How would you describe the one to one relationship between people and toothbrushes?
+1. How would you describe the many to many relationship between people and toothbrushes?
```

Comment:
This is a fun conversation waiting to happen.

Reply:
Wasn't sure if it was too out there of an example, seems like could spark some fun so going with it 😆